t/005-apicast-oauth.t

```diff
@@ -361,3 +362,38 @@ GET /?access_token=foobar
 --- error_code: 200
 --- response_body
 yay, upstream
+
+=== TEST 12: calling /authorize with state returns same value back on redirect_uri
```

Comment:
Yes, I can copy test 2 and 3, sending an additional state parameter, and they currently pass, but that won't test that the state that the client sends will be sent back to it with the code, since we need to invoke the callback to confirm this. 

That's what I'm struggling with, since to invoke callback I need to know what the apicast state value is, hence what test 12 is doing.

So my questions here would be whether test 12 is valid and whether we additionally need copies of tests 2 and 3 to validate that sending the state parameter doesn't break anything. 

Alternatively should test 12 be replaced with something else?

Where I'm struggling is in captur